### ISSUE TITLE
Mono build: OSX: Cannot run any scene

### ISSUE THREAD

**Issue by AlexeyGrishin** [godotengine/godot#24544]
**Godot version:**
v3.0.6.stable.mono.official.8314054

**OS/device including version:**
OSX

**Issue description:**
When trying to run any scene or open any C# script the editor does nothing, and `top` shows `Godot` process taking 62% CPU. 
<img width="179" alt="2018-12-22 13 39 10" src="https://user-images.githubusercontent.com/2125405/50373385-8187e400-05ef-11e9-9d23-cb5f95b0c73d.png">

**Steps to reproduce:**
1. Install Mono 5.12 (with MSBuild)
1. Install Godot build with Mono support
1. Create new project
1. Add Node2D to scene
1. Save & Run scene

Note that all works fine with non-Mono Godot build on my workstation. I suspect there is something wrong with Mono integration, but I don't know where to find logs or other debug information.

**Comment by kvonspiczak** [comment]
I'm experiencing the same issue on my macbook pro running OSX Mojave 10.14.1. Currently trying different mono versions to see if that changes anything.

**Comment by asciifaceman** [comment]
Same issue, OSX Sierra 10.12.6

Godot Godot_v3.0.6-stable_mono_osx 8314054

No debug output

I get the same behavior in a non-mono build but with this debug output:

```
** Debug Process Started **
Remote Debugger: Connection failed with status: '2', retrying in 1 msec.
OpenGL ES 3.0 Renderer: AMD Radeon Pro 560 OpenGL Engine
```

**Comment by AlexeyGrishin** [comment]
BTW, I see the same debug output in non-mono version, which works well, so I don't think it relates to issue.

**Comment by asciifaceman** [comment]
I was able to get non-mono version working by tweaking some settings - the play window was spawning with a size of 0x0

Mono version is still absolutely broken however, no window spawns but a process launches that slowly consumes memory + CPU .

**Comment by shianiawhite** [comment]
I fixed the problem in my case. Even though I had installed Mono 5.12 and Godot appeared to be finding this Mono installation, my other Mono installation through Homebrew (Mono 5.18) seemed to be interfering in some way. I uninstalled my Homebrew Mono and now Godot loads the scenes correctly. I should note, that when only Mono 5.18 was installed (before installing Mono 5.12) Godot would not even start. After installing Mono 5.12, Godot started but would not run any scene (so Godot seemed to be finding Mono 5.12). Then either the Homebrew Mono 5.18 is conflicting with the Mono 5.12 generally (which would be Homebrew's issue), or Godot was finding the Mono 5.12, but then using the Mono 5.18 anyway.

**Comment by jrowny** [comment]
> I was able to get non-mono version working by tweaking some settings - the play window was spawning with a size of 0x0

thanks for that, had to set test Test Width and Test Height, they were set to 0x0

**Comment by akien-mga** [comment]
Assuming fixed in 3.1 and later, as we didn't get further reports about this. There are no plans to release new Mono builds for the 3.0.x branch (which was early alpha for C# support).
